test/filterprov.c

```diff
@@ -167,7 +167,7 @@ int filter_provider_set_filter(int operation, const char *filterstr)
     if (globs->num_dispatch >= MAX_FILTERS)
         goto err;
 
-    for (name = filterstrtmp; !last; name = sep + 1) {
+    for (name = filterstrtmp; !last; name = (last ? NULL : sep + 1)) {
```

Comment:
That expression had me scratch my head and wondering what it was all about.  Suggested alternative: `(sep == NULL ? NULL : sep + 1)`

Reply:
Done